platform/jewel/ui/src/main/kotlin/org/jetbrains/jewel/ui/component/Popup.kt

```diff
@@ -18,7 +21,89 @@ import androidx.compose.ui.unit.IntOffset
 import androidx.compose.ui.unit.IntRect
 import androidx.compose.ui.unit.IntSize
 import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.window.Popup as ComposePopup
 import androidx.compose.ui.window.PopupPositionProvider
+import androidx.compose.ui.window.PopupProperties
+import org.jetbrains.jewel.foundation.InternalJewelApi
+import org.jetbrains.jewel.foundation.JewelConfigs
+import org.jetbrains.jewel.ui.component.PopupRender.PopupData
+
+/**
+ * Displays a popup with the provided content at a position determined by the given [PopupPositionProvider].
+ *
+ * This function behavior is influenced by the 'jewel.customPopupRender' system property. If set to `true`, it allows
+ * using a custom popup rendering implementation; otherwise, it defaults to the standard Compose popup.
+ *
+ * If running on the IntelliJ Platform and setting the 'jewel.customPopupRender' property to `true`, the plugin will use
+ * the JBPopup implementation for rendering popups. This is useful if your composable content is small and you need to
+ * display a popup that is bigger than the component size.
```

Comment:
I'd refer to the flags API here, instead of the raw property. Once we add KDocs to the flags API, people can refer to the one place where it's written.